packages/parser/utils/extractTextChunksFromPDF.ts

```diff
@@ -16,8 +17,24 @@ export async function extractTextChunksFromPDF(
       'pdfjs-dist/build/pdf.worker.entry'
     )
   }
+  
+  const documentProvider = getDocument(source);
+
+  let numberOfTries = 0;
+
+  documentProvider.onPassword = (
+    updatePassword: (password: string) => void,
+    _: any
+  ) => {
+    if (++numberOfTries > MAX_NUMBER_OF_TRIES) {
+      alert(TOO_MANY_PASSWORD_ATTEMPTS)
```

Comment:
Would it be possible to do the `alert()` in the application layer, rather than in the Parser library? The expectation is that the parser would be usable in both **Node.js** and **Browser** environments krub.

Reply:
Yes, it's possible. I already move it to the application layer.  Pls help check it again.

Reply:
Thank you!